dockerworker/tarstream.go

```diff
@@ -0,0 +1,67 @@
+// +build docker
+
+package dockerworker
+
+import (
+	"archive/tar"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+// a tarRewriter is a function that given a tar.Header and io.Reader returns a
+// tar.Header and io.Reader to be written in place of the entry given.
+//
+// If a nil header is returned  the entry is skipped, if no transformation is
+// desired the header and reader given can be returned with any changes.
+type tarRewriter func(header *tar.Header, r io.Reader) (*tar.Header, io.Reader, error)
+
+// rewriteTarStream rewrites a tar-stream read from r and written to w.
+//
+// For each entry in the tar-stream the rewriter function is called, if it
+// returns an error then the process is aborted, if it returns a nil header the
+// entry is skipped, otherwise the returned header and body is rewritten to the
+// output tar-stream. Notice that rewriter can return the arguments given
+// in-order to let entries pass-through.
+func rewriteTarStream(r io.Reader, w io.Writer, rewriter tarRewriter) error {
```

Comment:
what do we use this for?

Reply:
This is used to rewrite manifests in the docker image. This function was stolen from taskcluster-worker.

Reply:
I should have been clearer. Why do we need to rewrite manifests? What are we using them for, and why might they need changing?